src/Microsoft.AspNetCore.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -324,6 +342,37 @@ private void BuildFallbackBlock(TagHelperAttributeList attributes, DefaultTagHel
             }
         }
 
+        private string GetAttributeValue(object value)
+        {
+            string stringValue;
+            var htmlEncodedString = value as HtmlEncodedString;
+            if (htmlEncodedString != null)
+            {
+                // Value likely came from an HTML context in the .cshtml file but may still contain double quotes
+                // since attribute could have been enclosed in single quotes.
+                stringValue = htmlEncodedString.Value;
+                if (stringValue.Contains("\""))
+                {
+                    stringValue = stringValue.Replace("\"", "&quot;");
+                }
+            }
+            else
+            {
+                var writer = StringWriter;
+                RazorPage.WriteTo(writer, HtmlEncoder, value);
```

Comment:
This `static` method isn't fitting so well in `RazorPage` but I haven't got a better place to throw it. Will leave as-is unless someone has a good suggestion.